src/main/kotlin/org/rust/lang/utils/RustEscapesUtils.kt

```diff
@@ -0,0 +1,89 @@
+package org.rust.lang.utils
+
+import com.intellij.psi.StringEscapesTokenTypes.*
+import org.rust.lang.core.lexer.RustEscapesLexer
+import org.rust.lang.core.psi.RustTokenElementTypes.STRING_LITERAL
+
+/**
+ * Unescape string escaped using Rust escaping rules.
+ */
+fun String.unescapeRust(unicode: Boolean = true, eol: Boolean = true): String {
+    val sb = StringBuilder(length)
+    val lexer = RustEscapesLexer(STRING_LITERAL, unicode, eol)
+    lexer.start(this)
+    while (lexer.tokenType != null) {
+        sb.append(when (lexer.tokenType) {
+            VALID_STRING_ESCAPE_TOKEN -> decodeEscape(lexer.tokenText)
+            else                      -> lexer.tokenText
+        })
+        lexer.advance()
+    }
```

Comment:
This while loops look a little to low level. Can you use a Kotlin's `Sequqnce` here (via `generateSequence` probably)?

The API I imagine is 

```
fun String.tokenize(): Sequence<Something should be here?> {
  val lexer = RustEscapesLexer(???)
  lexer.start(this)
}
```